pkg/api/defaults.go

```diff
@@ -119,6 +120,9 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpdate bool) {
 			o.KubernetesConfig.KubernetesImageBase = cloudSpecConfig.KubernetesSpecConfig.KubernetesImageBase
 		}
 		if o.KubernetesConfig.EtcdVersion == "" || isUpdate {
+			if isUpdate && o.KubernetesConfig.EtcdVersion != DefaultEtcdVersion {
+				log.Warnf("etcd will be upgraded to version %s\n", DefaultEtcdVersion)
```

Comment:
this might be a downgrade (eg. If I set my etcdVersion to 3.3.x in my original deployment). I wonder if changing "upgraded" to something more general like "updated" makes sense. What do you think?

Reply:
Or if we want to be fancy we can put a conditional in there:

```
if isUpdate && IsVersionGt(o.KubernetesConfig.EtcdVersion, DefaultEtcdVersion) {
     log.Warnf("etcd will be downgraded to version %s\n", DefaultEtcdVersion)
} else if isUpdate && IsVersionLt(o.KubernetesConfig.EtcdVersion, DefaultEtcdVersion) {
     log.Warnf("etcd will be upgraded to version %s\n", DefaultEtcdVersion)
}

```

Reply:
Hmm, there's also the question of the mixed meaning of `isUpdate`. Will that be true in a scale scenario as well? In which case this message is misleading, as scale never effects the control plane and thus etcd will not be effected.

Reply:
`isUpdate == isUpgrade || isScale`

That's a good point. In that case we might want to replace `isUpdate` by `isUpgrade`.